File: Java/alarms-randori/src/main/java/AlarmServices/AlarmService.java
package AlarmServices;

import Main.Alarm;

public class AlarmService {
    public static void triggerAlarm(Alarm alarm) {
      switch (alarm.alarmTypes) {
        case Loud:
          LoudAlarmService.trigger();
          break;
        case Silent:
          SilentAlarmService.trigger(alarm);
          break;
        case Composite:
          CompositeAlarmService.trigger(alarm);
          break;
        case Timed:
          TimedAlarmService.trigger(alarm);
          break;
        default:
          throw new IllegalArgumentException("out or range");
      }
    }

    public static void disableAlarm(Alarm alarm) {
      switch (alarm.alarmTypes) {
        case Loud:
          LoudAlarmService.disable();
          break;
        case Silent:
          SilentAlarmService.disable(alarm);
          break;
        case Composite:
          CompositeAlarmService.disable(alarm);
          break;
        case Timed:
          TimedAlarmService.disable(alarm);
          break;
        default:
          throw new IllegalArgumentException("out or range");
      }
    }

    public static void dump(Alarm alarm) {
      switch (alarm.alarmTypes) {
        case Loud:
          LoudAlarmService.dump(alarm);
          break;
        case Silent:
          SilentAlarmService.dump(alarm);
          break;
        case Composite:
          CompositeAlarmService.dump(alarm);
          break;
        case Timed:
          TimedAlarmService.dump(alarm);
          break;
        default:
          throw new IllegalArgumentException("out or range");
      }
    }
  }

File: TellDontAsk/AlarmsProcedural/AlarmServices/AlarmService.cs
using System;
using AlarmsProcedural.Enums;

namespace AlarmsProcedural.AlarmServices
{
  public static class AlarmService
  {
    public static void TriggerAlarm(Alarm alarm)
    {
      switch (alarm.AlarmType)
      {
        case AlarmTypes.Loud:
          LoudAlarmService.Trigger();
          break;
        case AlarmTypes.Silent:
          SilentAlarmService.Trigger(alarm);
          break;
        case AlarmTypes.Composite:
          CompositeAlarmService.Trigger(alarm);
          break;
        case AlarmTypes.Timed:
          TimedAlarmService.Trigger(alarm);
          break;
        default:
          throw new ArgumentOutOfRangeException();
      }
    }

    public static void DisableAlarm(Alarm alarm)
    {
      switch (alarm.AlarmType)
      {
        case AlarmTypes.Loud:
          LoudAlarmService.Disable();
          break;
        case AlarmTypes.Silent:
          SilentAlarmService.Disable(alarm);
          break;
        case AlarmTypes.Composite:
          CompositeAlarmService.Disable(alarm);
          break;
        case AlarmTypes.Timed:
          TimedAlarmService.Disable(alarm);
          break;
        default:
          throw new ArgumentOutOfRangeException();
      }
    }

    public static void Dump(Alarm alarm)
    {
      switch (alarm.AlarmType)
      {
        case AlarmTypes.Loud:
          LoudAlarmService.Dump();
          break;
        case AlarmTypes.Silent:
          SilentAlarmService.Dump(alarm);
          break;
        case AlarmTypes.Composite:
          CompositeAlarmService.Dump(alarm);
          break;
        case AlarmTypes.Timed:
          TimedAlarmService.Dump(alarm);
          break;
        default:
          throw new ArgumentOutOfRangeException();
      }
    }
  }
}